src/ProtectedProxies/MinimalProtectedProxyFactory.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: UNLICENSED
+// (c) SphereX 2023 Terms&Conditions
+
+pragma solidity ^0.8.0;
+
+import {SphereXConfiguration} from "../SphereXConfiguration.sol";
+import {SpherexProtetedMinimalProxy} from "./ProtectedMinimalProxy.sol";
+
+// TODO: add event for deploying
+
+/**
+ * @title A factory contract that deploys the SpherexProtetedMinimalProxy contracts.
+ */
+contract MinimalProtectedProxyFactory is SphereXConfiguration {
+
+    address private immutable IMPLEMENTATION;
+    address private immutable ALLOWED_DEPLOYER;
+    bytes4[] private /*immutable*/ PROTECTED_SIGS; // this is in effect immutable.
```

Comment:
in effect? in fact?

Reply:
No, it's in effect. Effecitvly its immutalbe (the is no way to mutate it) but it not `immutable` in the sense that it's a variable and not a const in the bytecode.